Plutarch/DataRepr.hs

```diff
@@ -35,60 +38,67 @@ import Plutarch.Builtin (
   pfstBuiltin,
   psndBuiltin,
  )
+import Plutarch.Field.HList (type IndexList)
 import Plutarch.Integer (PInteger)
 import Plutarch.Lift (PConstant, PConstantRepr, PConstanted, PLift, pconstantFromRepr, pconstantToRepr)
 import Plutarch.List (punsafeIndex)
 import Plutarch.Prelude
 import qualified Plutus.V1.Ledger.Api as Ledger
 import qualified PlutusCore as PLC
 
-data PDataList (as :: [PType]) (s :: S)
+data PDataRecord (as :: [PLabeled]) (s :: S)
 
-pdhead :: Term s (PDataList (a : as) :--> PAsData a)
+data PLabeled = Symbol := PType
+
+pdhead :: Term s (PDataRecord ((l ':= a) : as) :--> PAsData a)
 pdhead = phoistAcyclic $ pforce $ punsafeBuiltin PLC.HeadList
```

Comment:
I wonder if we want more operations for working with `PDataRecord`s, or if all of this is sufficient :thinking: